app/victoria-metrics/main_test.go

```diff
@@ -0,0 +1,270 @@
+// +build integration
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert"
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect"
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vmstorage"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+const (
+	testFixturesDir        = "testdata"
+	testStorageSuffix      = "vm-test-storage"
+	testHTTPListenAddr     = ":7654"
+	testStatsDListenAddr   = ":2003"
+	testOpenTSDBListenAddr = ":4242"
+)
+
+const (
+	testWaitForDataTimeout = 5 * time.Second
+	testStorageInitTimeout = 30 * time.Second
+)
+
+var (
+	storagePath   string
+	insertionTime = time.Now().UTC()
+)
+
+type test struct {
+	Name   string `json:"name"`
+	Data   string `json:"data"`
+	Query  string `json:"query"`
+	Result []Row  `json:"result"`
+}
+
+type Row struct {
+	Metric     map[string]string `json:"metric"`
+	Values     []float64         `json:"values"`
+	Timestamps []int64           `json:"timestamps"`
+}
+
+func TestMain(m *testing.M) {
+	setUp()
+	code := m.Run()
+	tearDown()
+	os.Exit(code)
+}
+
+func setUp() {
+	storagePath = filepath.Join(os.TempDir(), testStorageSuffix)
+	processFlags()
+	logger.Init()
+	vmstorage.Init()
+	vmselect.Init()
+	vminsert.Init()
+	go httpserver.Serve(*httpListenAddr, requestHandler)
+	readyStorageCheckFunc := func() bool {
+		resp, err := http.Get("http://127.0.0.1" + testHTTPListenAddr + "/health")
```

Comment:
No need to concat `"http://127.0.0.1" + testHTTPListenAddr` - just set it to flag:
```
	testHTTPListenAddr     = "http://127.0.0.1:7654"

```

Reply:
testHTTPListenAddr is used in processFlags func

Reply:
Oh well, that flag supposed to be set without scheme. Right, still feeling weird from doing `"http://127.0.0.1" + testHTTPListenAddr` at least 5 times.